src/Analyzers/CSharp/Analyzers/UseImplicitObjectCreation/CSharpUseImplicitObjectCreationDiagnosticAnalyzer.cs

```diff
@@ -93,18 +98,61 @@ private void AnalyzeSyntax(SyntaxNodeAnalysisContext context)
                     _ => null,
                 };
             }
+            else if (objectCreation.Parent.IsKind(SyntaxKind.ArrayInitializerExpression) &&
+                objectCreation.Parent.Parent is ArrayCreationExpressionSyntax arrayCreation)
+            {
+                typeNode = arrayCreation.Type.ElementType;
+            }
+            else if (objectCreation.Parent.IsKind(SyntaxKind.CollectionInitializerExpression) &&
+                objectCreation.Parent.Parent is ObjectCreationExpressionSyntax collectionObjectCreation)
+            {
+                var collectionTypeSymbol = semanticModel.GetTypeInfo(collectionObjectCreation, cancellationToken).Type!;
+                var targetTypeSymbol = semanticModel.GetTypeInfo(objectCreation, cancellationToken).ConvertedType!;
+                var argumentTypeSymbols = new[] { targetTypeSymbol }.ToImmutableList();
```

Comment:
Can this be `ImmutableList.Create(targetTypeSymbol)`?

Reply:
Actually immutable here is completely unnecessary. Changed to a simple list.